File: src/main/java/io/github/robaertschi/minetech/item/ModItems.java
package io.github.robaertschi.minetech.item;

import io.github.robaertschi.minetech.Minetech;
import net.minecraft.world.item.Item;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;

public class ModItems {


    public static final DeferredRegister<Item> ITEMS = DeferredRegister.create(ForgeRegistries.ITEMS, Minetech.MODID);



    public static void register(IEventBus eventBus) {
        ITEMS.register(eventBus);
    }

}

File: src/main/kotlin/robaertschi/minetech/item/ModItems.kt
package robaertschi.minetech.item

import net.minecraft.world.item.CreativeModeTab
import net.minecraft.world.item.Item
import net.minecraftforge.registries.DeferredRegister
import net.minecraftforge.registries.ForgeRegistries
import robaertschi.minetech.Minetech
import thedarkcolour.kotlinforforge.forge.registerObject

object ModItems {

    val REGISTRY: DeferredRegister<Item> = DeferredRegister.create(ForgeRegistries.ITEMS, Minetech.ID);

    val EXAMPLE_ITEM by REGISTRY.registerObject("example_item") {
        Item(Item.Properties().tab(CreativeModeTab.TAB_MISC))
    }


}